Sources/TuistKit/Services/TestService.swift

```diff
@@ -36,47 +37,46 @@ enum TestServiceError: FatalError {
 
 final class TestService {
     /// Project generator
-    let generator: Generating
-
-    /// Xcode build controller.
-    let xcodebuildController: XcodeBuildControlling
-
-    /// Build graph inspector.
-    let buildGraphInspector: BuildGraphInspecting
-
-    /// Simulator controller
-    let simulatorController: SimulatorControlling
+    private let generatorInit: (AbsolutePath) -> Generating
+    private let xcodebuildController: XcodeBuildControlling
+    private let buildGraphInspector: BuildGraphInspecting
+    private let simulatorController: SimulatorControlling
 
     private let temporaryDirectory: TemporaryDirectory
+    private let testsCacheTemporaryDirectory: TemporaryDirectory
 
-    convenience init(
-        xcodebuildController _: XcodeBuildControlling = XcodeBuildController(),
-        buildGraphInspector _: BuildGraphInspecting = BuildGraphInspector(),
-        simulatorController _: SimulatorControlling = SimulatorController()
-    ) throws {
+    convenience init() throws {
         let temporaryDirectory = try TemporaryDirectory(removeTreeOnDeinit: true)
+        let testsCacheTemporaryDirectory = try TemporaryDirectory(removeTreeOnDeinit: true)
         self.init(
             temporaryDirectory: temporaryDirectory,
-            generator: Generator(
-                projectMapperProvider: AutomationProjectMapperProvider(),
-                graphMapperProvider: GraphMapperProvider(),
-                workspaceMapperProvider: AutomationWorkspaceMapperProvider(
-                    workspaceDirectory: temporaryDirectory.path
-                ),
-                manifestLoaderFactory: ManifestLoaderFactory()
-            )
+            testsCacheTemporaryDirectory: testsCacheTemporaryDirectory,
+            generatorInit: { automationPath in
+                Generator(
+                    projectMapperProvider: AutomationProjectMapperProvider(),
+                    graphMapperProvider: AutomationGraphMapperProvider(
+                        testsCacheDirectory: testsCacheTemporaryDirectory.path
+                    ),
+                    workspaceMapperProvider: AutomationWorkspaceMapperProvider(
+                        workspaceDirectory: FileHandler.shared.resolveSymlinks(automationPath)
+                    ),
+                    manifestLoaderFactory: ManifestLoaderFactory()
+                )
+            }
         )
     }
 
     init(
         temporaryDirectory: TemporaryDirectory,
-        generator: Generating,
+        testsCacheTemporaryDirectory: TemporaryDirectory,
+        generatorInit: @escaping (AbsolutePath) -> Generating,
```

Comment:
does this bring any value over a more lightweight closure? I can make a `typealias` if typealias is what you are worried about. Otherwise, I don't see really what this brings but maybe I am missing something? And with a protocol I'll have to create an implementation class, mock - that's a lot of boilerplate for just a bunch of lines of code

Reply:
It's mainly being consistent with the patterns we've been using across the codebase. We achieve the same thing, but we are consistent, which at the end of the day is very appreciated when working on a large codebase. We can also provide mocks by conforming the protocol of that factory.

Reply:
that's a good point, I'll refactor it 👍